src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -84,7 +90,12 @@
   private PreparedStatement deleteRepairScheduleByClusterAndKsPrepStmt;
 
   public CassandraStorage(ReaperApplicationConfiguration config, Environment environment) {
-    cassandra = config.getCassandraFactory().build(environment).register(QueryLogger.builder().build());
+    
+    CassandraFactory cassandraFactory = config.getCassandraFactory();
+    // all INSERT and DELETE statement prepared in this class are idempotent
+    cassandraFactory.setQueryOptions(java.util.Optional.of(new QueryOptions().setDefaultIdempotence(true)));
```

Comment:
retry and speculative executions apply to writes if idempotent is true. in theory at least, i'm not sure of the current implementation of the driver. regardless it's worth stating it for clarity on design.